### PR TITLE
swap min to hour

### PR DESCRIPTION
**Pull request by justinzhou93**

### CODE REVIEW

**Comment by justinzhou93** [comment]
@lreesby should be a good base to work off of. should be mostly done, unit tests and benchmarks and all. i will run QA testing tomorrow morning, but otherwise, should be good to go

**Comment by justinzhou93** [comment]
QA'ed following scenarios:
1. migration removes all existing tasks from all queues and maps
2. storage version bumps to version 2 (storage version exists separately for each pallet)
3. last time slot is set to the current hour slot for both last time slot and last missed time slot, before it continues forward. similarly to before, the last missed time slot lags by 1 slot, so in this case, 3600 seconds.
4. errors when don't schedule on the hour
5. 60 transfers per block, but will complete many over the course of many blocks throughout the hour (max set to 1000 right now)

File: pallets/automation-time/src/migrations.rs [hunk 1]
```diff
@@ -23,3 +23,41 @@ pub mod v1 {
...
+	use crate::{LastTimeSlot, Pallet, Task, Vec, TaskQueue, MissedQueue};
+
+	use super::*;
+
+	pub fn migrate<T: Config>() -> Weight {
+		info!(target: "automation-time", "Migrating automation-time v2");
+		let pallet_prefix: &[u8] = b"AutomationTime";
+
+		let scheduled_tasks_prefix: &[u8] = b"ScheduledTasks";
+		let _scheduled_tasks: Vec<_> = storage_key_iter::<u64, BoundedVec<T::Hash, T::MaxTasksPerSlot>, Twox64Concat>(pallet_prefix, scheduled_tasks_prefix)
```
Comment (rhuttman) [inline pallets/automation-time/src/migrations.rs:39]: how does this work

File: pallets/automation-time/src/tests.rs [hunk 2]
```diff
@@ -785,6 +785,49 @@ fn on_init_shutdown() {
...
 }
 
+#[test]
+fn migration_v1() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		AutomationTime::on_initialize(1);
+		let new_timestamp = START_BLOCK_TIME + (3600 * 1_000);
+		Timestamp::set_timestamp(new_timestamp);
+		v1::migrate::<Test>();
+		if let Some((updated_last_time_slot, updated_last_missed_slot)) =
```
Comment (rhuttman) [inline pallets/automation-time/src/tests.rs:795]: you don't have a fail in there for the else

File: pallets/automation-time/src/tests.rs [hunk 3]
```diff
@@ -785,6 +785,49 @@ fn on_init_shutdown() {
...
+
+		v2::migrate::<Test>();
+		Timestamp::set_timestamp(START_BLOCK_TIME + (3600 * 1_000));
+		if let Some((updated_last_time_slot, updated_last_missed_slot)) =
+			AutomationTime::get_last_slot()
+		{
+			assert_eq!(updated_last_time_slot, LAST_BLOCK_TIME,);
+			assert_eq!(updated_last_missed_slot, LAST_BLOCK_TIME,);
+		}
+		match AutomationTime::get_scheduled_tasks(SCHEDULED_TIME) {
```
Comment (rhuttman) [inline pallets/automation-time/src/tests.rs:820]: also need to check the Tasks map

**Review by rhuttman — CHANGES_REQUESTED** [review]

File: pallets/automation-time/src/migrations.rs (hunk 1 above)
Comment (justinzhou93) [reply pallets/automation-time/src/migrations.rs:39]: the .drain() is a function on an iterator, which removes everything from the storage map  https://docs.substrate.io/rustdocs//latest/src/frame_support/storage/migration.rs.html#61

and storage_key_iter just provides the iterator to drain the scheduledTasks storage: https://docs.substrate.io/rustdocs//latest/src/frame_support/storage/migration.rs.html#193

was following moonbeam here: https://github.com/PureStake/moonbeam/blob/e7b414a1370f7af5705e31835336460181be9267/pallets/asset-manager/src/migrations.rs#L82

File: pallets/automation-time/src/tests.rs (hunk 3 above)
Comment (justinzhou93) [reply pallets/automation-time/src/tests.rs:824]: gahh i knew i forgot something on this test

File: runtime/neumann/src/lib.rs [hunk 4]
```diff
@@ -212,7 +212,7 @@ pub const fn deposit(items: u32, bytes: u32) -> Balance {
 pub const EXISTENTIAL_DEPOSIT: Balance = DOLLAR / 10;
 
 /// We use at most 10% of the block weight running scheduled tasks during `on_initialize`.
-const SCHEDULED_TASKS_INITIALIZE_RATIO: Perbill = Perbill::from_percent(10);
+const SCHEDULED_TASKS_INITIALIZE_RATIO: Perbill = Perbill::from_percent(5);
```
Comment (rhuttman) [inline runtime/neumann/src/lib.rs:215]: also do for TUR

**Review by rhuttman — CHANGES_REQUESTED** [review]

**Review by rhuttman — APPROVED** [review]
lgtm

**Review by rhuttman — APPROVED** [review]
lgtm

**Pull request merged**